README_cmake.md

```diff
@@ -2,226 +2,287 @@
 
 ## Overview
 
-The primary support for Halide in CMake is the
-[`halide_library`](#halide_library) rule, which allows you to build a Generator
-into an executable for various architectures.
+There are two main ways to use Halide in your application: as a **JIT compiler**
+for dynamic pipelines or an **ahead-of-time (AOT) compiler** for static pipelines.
 
-Halide code is produced in a two-stage build that may seem a bit unusual at
-first, so before documenting the build rules, let's recap: a *Generator* is a
-C++ class in which you define a Halide pipeline and schedule, as well as
-well-defined input and output parameters; it can be compiled into an executable
-(often referred to as a *Filter*) that efficiently runs the Halide pipeline.
+CMake provides robust support for both use cases. These new features rely on recent
+developments in CMake, so a recent version is necessary. In order to build Halide,
+you would ideally have the newest version of CMake installed on your system (3.17
+at time of writing). However, everything should work with version 3.16 or newer.
```

Comment:
Be specific here as to what the minimum CMake version required is (not just "should work")